### ISSUE TITLE
how to root???

### ISSUE THREAD

**Issue by Misschihiro** [Lunars/tesla#70]
hi all, the only prerequisite is the muc has been rooted, but how to root the MCU???
![image](https://user-images.githubusercontent.com/17743767/99932061-d01cfa80-2d91-11eb-9780-ac574a15deef.png)

**Comment by it-gemini** [comment]
Good question :) Did you do it?

**Comment by Randvegeta** [comment]
Yup. Rooted

**Comment by lau85** [comment]
Rooted with what firmware version?
It was super easy few months ago, but they have added some changes with latest versions.

**Comment by Randvegeta** [comment]
Upto 2020.44

2020.48 no longer uses the tesla1 token. Need another way.

**Comment by GENOV7** [comment]
I am very excited about 2020.48

**Comment by vacekj** [comment]
Would anyone be willing to point in the right direction to rooting and flashing a Model 3 on latest firmware?

**Comment by stiffler-m** [comment]
> 
> 
> Would anyone be willing to point in the right direction to rooting and flashing a Model 3 on latest firmware?

BIOS

**Comment by Anton30Z** [comment]
Hello
How to root on the last firmware????

**Comment by stiffler-m** [comment]
> Hello How to root on the last firmware????


it is difficult to get root on a recent firmware

**Comment by vdiravka** [comment]
I have a full access to IC. I can ping CID and GW from there. But ssh too CID is disabled:
```
################################################################################
# Config: /etc/ssh/sshd_config_prod                                            #
# Date: Tue Dec  5 11:16:33 PST 2023                                           #
# Authorized principals:                                                       #
#   tesla:motors:vehicle:5YJSA1E22GF1XXXXX                                     #
#   tesla:motors:vehicle:5YJSA1E22GF1XXXXX                                     #
#                                                                              #
# SSH allowed: vehicle parked                                                  #
################################################################################
tesla1@cid: Permission denied (publickey)
```
Any tips how to bypass?

About gw, it's a pity that diagnostic port closed and doesn't accept command `printf "\x12\x01" |socat - udp:gw:3500` to open 23 shell port

**Comment by strmotors** [comment]
> I have dull access to IC. I can ping CID and GW from there. But ssh too CID is disabled:
> 
> ```
> ################################################################################
> # Config: /etc/ssh/sshd_config_prod                                            #
> # Date: Tue Dec  5 11:16:33 PST 2023                                           #
> # Authorized principals:                                                       #
> #   tesla:motors:vehicle:5YJSA1E22GF1XXXXX                                     #
> #   tesla:motors:vehicle:5YJSA1E22GF1XXXXX                                     #
> #                                                                              #
> # SSH allowed: vehicle parked                                                  #
> ################################################################################
> tesla1@cid: Permission denied (publickey)
> ```
> 
> Any tips how to bypass?
> 
> About gw, it's a pity that diagnostic port closed and doesn't accept command `printf "\x12\x01" |socat - udp:gw:3500` to open 23 shell port

That was an old method of rooting, by copying ssh keys of ic to local server and rooting cid. They have fixed that around 2020. There is no new public method.

**Comment by vdiravka** [comment]
> That was an old method of rooting, by copying ssh keys of ic to local server and rooting cid. They have fixed that around 2020. There is no new public method.

agreed, looks like the main fix was in closing `printf "\x12\x01" |socat - udp:gw:3500` command and accessing Gatewey shell

**Comment by scsikid** [comment]
It's 2025, can anyone point to a guide using the latest mcu1 firmware to root? At the moment I just want to make my car a hotspot for my iPad ;)

**Comment by stiffler-m** [comment]
> It's 2025, can anyone point to a guide using the latest mcu1 firmware to root? At the moment I just want to make my car a hotspot for my iPad ;)

Of course. But it will cost you dearly.

**Comment by scsikid** [comment]
> > It's 2025, can anyone point to a guide using the latest mcu1 firmware to root? At the moment I just want to make my car a hotspot for my iPad ;)
> 
> Of course. But it will cost you dearly.

Ok what is the cost? @stiffler-m

**Comment by sebastianpc** [comment]
@stiffler-m any ways to contact you?
Otherwise please reach out to me: tesla@sebastianpc.me
